tests/functional/apps/zilliqa.py

```diff
@@ -64,9 +65,14 @@ def parse_get_public_key_response(self, response: bytes) -> (bytes, str, bytes):
 
     @contextmanager
     def send_async_get_public_key(self, index: int,
-                                  disp_addr: bool) -> Generator[None, None, None]:
+                                  disp_addr: bool,
+                                  verify: bool = True) -> Generator[None, None, None]:
         p1 = 0
         p2 = P2_DISPLAY_ADDRESS if disp_addr else P2_DISPLAY_PUBKEY
+
+        if not verify:
+            p2 |= P2_DISPLAY_NO_VERIFY
+
```

Comment:
I think it would be better to create another method which would be something like:
```
    def send_get_public_key_non_confirm(self, index: intl) -> RAPDU:
        p1 = 0
        p2 = P2_DISPLAY_NO_VERIFY
        payload = pack("<I", index)
        return self._client.exchange(CLA, INS.INS_GET_PUBLIC_KEY,
                                     p1, p2, payload)
```
So that it doesn't get used with navigation instructions as there is no UX flow for this call.

Reply:
Yes, I can add that. Note at that `P2_DISPLAY_NO_VERIFY` is a flag rather than a value in its own right. Values are:

- `P2_DISPLAY_PUBKEY = 0x00`
- `P2_DISPLAY_ADDRESS = 0x01`
- `P2_DISPLAY_PUBKEY | P2_DISPLAY_NO_VERIFY = 0x02`
- `P2_DISPLAY_ADDRESS | P2_DISPLAY_NO_VERIFY = 0x03`

Note that you can request either the public key or the address without asking user verification on the UI. Using a flag rather than separate parameters are somewhat aligned with how it is implemented Ledger Live Desktop where verifying the address on the hardware wallet is a flag (`hw-getAddress.ts`):

```typescript
const resolver: Resolver = async (transport, { path, verify }) => {
  const zilliqa = new Zilliqa(transport);
  const r = await zilliqa.getAddress(path, !!verify);

  return {
    address: r.address,
    publicKey: r.publicKey,
    path,
  };
};
```

Reply:
I'm sorry I don't get what would be the difference between:
- P2_DISPLAY_PUBKEY | P2_DISPLAY_NO_VERIFY = 0x02
- P2_DISPLAY_ADDRESS | P2_DISPLAY_NO_VERIFY = 0x03
There won't be any user UI on the device side, so no change here.
And the response contains both the public key and the address in both cases.

Therefore I don't get why you should handle this as a flag rather than as a third option in an enum?

Reply:
You are completely right. This is a brainfart on my part: There should only be one flag as the comms always return both address and public key.

Reply:
That should be all fixed now.